numba/typing/builtins.py

```diff
@@ -866,3 +877,65 @@ class DeferredAttribute(AttributeTemplate):
 
     def generic_resolve(self, deferred, attr):
         return self.context.resolve_getattr(deferred.get(), attr)
+
+
+#------------------------------------------------------------------------------
+
+
+@infer
+class CmpEq(AbstractTemplate):
```

Comment:
I'm not sure this is necessary. Or, if it is, you should remove all other typing declarations for `==`.